custom_components/tahoma/coordinator.py

```diff
@@ -50,16 +57,42 @@ async def _async_update_data(self) -> Dict[str, Device]:
             raise UpdateFailed(f"Error communicating with the TaHoma API: {exception}")
         else:
             for event in events:
+                _LOGGER.debug(
+                    f"{event.name}/{event.exec_id} (device:{event.deviceurl},state:{event.old_state}->{event.new_state})"
+                )
+
+                if event.name == "DeviceAvailableEvent":
+                    self.devices[event.deviceurl].available = True
+
                 if event.name == "DeviceStateChangedEvent":
                     for state in event.device_states:
                         device = self.devices[event.deviceurl]
                         if state.name not in device.states:
                             device.states[state.name] = state
                         device.states[state.name].value = self._get_state(state)
+
+                if (
+                    event.name == "ExecutionRegisteredEvent"
+                    and event.exec_id in self.executions
+                ):
+                    self.update_interval = timedelta(seconds=1)
+
+                if (
+                    event.name == "ExecutionStateChangedEvent"
+                    and event.exec_id in self.executions
```

Comment:
I don't what to think about this. In our tahoma library, I now update the device state and recreate the session to solve the disconnect session error we met: https://github.com/iMicknl/python-tahoma-api/pull/36/files#diff-f6a3053f6cfdb465b9897b1a29772ea4R96
The linked code will be not compatible with your's :S

Reply:
To be honest, I'd rather keep the update at a high frequency all the time. I use the tahoma app quite often and I'd like the state to be reflected promptly in HA rather than wait for an update. Plus according to the tahoma doc on event listener, it seems to be built to have fetches every seconds. IMHO having the reconnect in case of a disconnection is a better solution.

Reply:
@vlebourl in that case you are able to set the event listener to 1 second via the options? I can understand that there are users who are not comfortable with having a request every second. And I am not sure if that will even be accepted in core, as a default...

This change will make sure that users are able to configure the default polling interval, thus they can choose to instant (every second) and 'controlled by HA only' and set it to an hour or more. However, in case of a long polling interval (like our current default), updates will be very slow after taking an action from HA, which should not be normal behaviour in my opinion.

Reply:
fair enough. Is there a way to reconcile @tetienne 's code though?

Reply:
> I don't what to think about this. In our tahoma library, I now update the device state and recreate the session to solve the disconnect session error we met: [iMicknl/python-tahoma-api#36 (files)](https://github.com/iMicknl/python-tahoma-api/pull/36/files#diff-f6a3053f6cfdb465b9897b1a29772ea4R96)
> The linked code will be not compatible with your's :S

Let's not merge this PR until we figure out the best way! I can understand your approach, however I believe this will make the package less flexible. My preference would be to try to auto reconnect and throw a certain status, so that we can retrieve the latest device states with a force refresh in the HA integration. 

Did we try to see if our current event listener can still be retrieved, after an error? We could re login automatically and see if we can pull the same event listener.

Reply:
I like that approach

Reply:
I didn't try to recall the same listener once logout and re-login. But's it's safer to recreate it.

Look at the Hue integration, they set the poll interval to 5 seconds: https://github.com/home-assistant/core/blob/dev/homeassistant/components/hue/light.py#L38

So we should be able to do the same.